pkg/storage/errors.go

```diff
@@ -68,15 +70,24 @@ func NewUnreachableError(key string, rv int64) *StorageError {
 	}
 }
 
+func NewInvalidObjError(key, msg string) *StorageError {
+	return &StorageError{
+		Code:               ErrCodeInvalidObj,
+		Key:                key,
+		AdditionalErrorMsg: msg,
+	}
+}
+
 type StorageError struct {
-	Code            int
-	Key             string
-	ResourceVersion int64
+	Code               int
+	Key                string
+	ResourceVersion    int64
+	AdditionalErrorMsg string
```

Comment:
So the errors being returned come directly from the source and are not transformed.  What is your opinion on transforming them such that they are all storage errors?